core-api/README.md

```diff
@@ -0,0 +1,102 @@
+
+### RaiseExceptionErrorHandler
+[`RaiseExceptionErrorHandler`](https://github.com/optimizely/java-sdk/blob/master/core-api/src/main/java/com/optimizely/ab/error/RaiseExceptionErrorHandler.java)
+
+## EventProcessor
+[`EventProcessor`](https://github.com/optimizely/java-sdk/blob/master/core-api/src/main/java/com/optimizely/ab/event/EventProcessor.java) 
+interface is used to provide an intermediary processing stage within event production.
+It's assumed that the `EventProcessor` dispatches events via a provided `EventHandler`.
+
+### BatchEventProcessor
+[`BatchEventProcessor`](https://github.com/optimizely/java-sdk/blob/master/core-api/src/main/java/com/optimizely/ab/event/BatchEventProcessor.java)
+is a batched implementation of the `EventProcessor`. Events passed to the BatchEventProcessor are immediately added to a BlockingQueue.
+The BatchEventProcessor maintains a single consumer thread that pulls events off of the BlockingQueue and buffers them for either a 
+configured batch size or for a maximum duration before the resulting `LogEvent` is sent to the `EventDispatcher` and `NoticifactionCenter`.
+
+### ForwardingEventProcessor
+[`ForwardingEventProcessor`](https://github.com/optimizely/java-sdk/blob/master/core-api/src/main/java/com/optimizely/ab/event/ForwardingEventProcessor.java)
+implements `EventProcessor` for backwards compatibility. Each event Processed is translated into a [`LogEvent`](https://github.com/optimizely/java-sdk/blob/master/core-api/src/main/java/com/optimizely/ab/event/ForwardingEventProcessor.java)
+before getting synchronously sent along to the supplied `EventHandler`.
+
+## EventHandler
+[`EventHandler`](https://github.com/optimizely/java-sdk/blob/master/core-api/src/main/java/com/optimizely/ab/event/EventHandler.java)
+interface is used for dispatching events to the Optimizely event end-point. Implementations of `dispatchEvent(LogEvent)` are expected
+to make an HTTP request of type `getRequestMethod()` to the `LogEvent#getEndpointUrl()` location. The corresponding request parameters and body
+are available via `LogEvent#getRequestParams()` and `LogEvent#getBody()` respectively.
+
+### NoopEventHandler
+[`NoopEventHandler`](https://github.com/optimizely/java-sdk/blob/master/core-api/src/main/java/com/optimizely/ab/event/NoopEventHandler.java)
+implements `EventHandler` with no side-effects. Useful for testing or non-production environments.
+
+## NotificationCenter
+[`NotificationCenter`](https://github.com/optimizely/java-sdk/blob/master/core-api/src/main/java/com/optimizely/ab/notification/NotificationCenter.java)
+is the centralized component for subscribing to notifications from the SDK. Subscribers must implement the `NotificationHandler<T>` interface
+and are registered via `NotificationCenter#addNotificationHandler`. Notifications are currently sent synchronously, so be sure that 
+implementations do not block unnecessarily.
+
```

Comment:
Suggest revising to specify what is blocked. Do you mean do not block the notifications? Also, "unnecessarily" sounds like it might be necessary to block in some circumstances

Reply:
also not sure how to revise.

Reply:
The sentence "Notifications are currently sent synchronously, so be sure that implementations do not block unnecessarily." isn't explicit about what the implementations could block. For example, do you mean "implementations do not block notifications unnecessarily" maybe? It seems like a noun is missing after "block"

The second question is that "unnecessarily" suggests that the possibility that there is a time to "necessarily" block notifications (or whatever you are suggesting they avoid blocking). So a reader might wonder -- what would be the case for necessarily blocking notifications (whatever you are suggesting they avoid blocking]

Does that make sense?